crates/fluvio-sc/src/services/public_api/mirroring/connect.rs

```diff
@@ -0,0 +1,169 @@
+
+use crate::core::Context;
+
+// This is the entry point for handling mirroring requests
+// Home clusters will receive requests from remote clusters
+pub struct MirroringConnectController<C: MetadataItem> {
+    req: MirrorConnect,
+    response_sink: ExclusiveFlvSink,
+    end_event: Arc<StickyEvent>,
+    ctx: Arc<Context<C>>,
+    header: RequestHeader,
+}
+
+const MIRRORING_CONTROLLER_INTERVAL: u64 = 5;
+
+impl<C: MetadataItem> MirroringConnectController<C> {
+    pub fn start(
+        req: MirrorConnect,
+        response_sink: ExclusiveFlvSink,
+        end_event: Arc<StickyEvent>,
+        ctx: Arc<Context<C>>,
+        header: RequestHeader,
+    ) {
+        let controller = Self {
+            req: req.clone(),
+            response_sink,
+            end_event,
+            ctx,
+            header,
+        };
+
+        spawn(controller.dispatch_loop());
+    }
+
+    #[instrument(skip(self), name = "MirroringConnectControllerLoop")]
+    async fn dispatch_loop(mut self) {
+        use tokio::select;
+        info!(name = self.req.name, "received mirroring connect request");
+
+        let mut topics_listener = self.ctx.topics().change_listener();
+
+        loop {
+            if self
+                .sync_and_send_topics(&mut topics_listener)
+                .await
+                .is_err()
+            {
+                self.end_event.notify();
+                break;
+            }
+
+            trace!("{}: waiting for changes", self.req.name);
+            select! {
+                _ = self.end_event.listen() => {
+                    debug!("connection has been terminated");
+                    break;
+                },
+
+                _ = topics_listener.listen() => {
+                    debug!("mirroring: {}, topic changes has been detected", self.req.name);
+                }
+            }
+
+            // sleep for a while
+            debug!("sleeping for {} seconds", MIRRORING_CONTROLLER_INTERVAL);
+            sleep(Duration::from_secs(MIRRORING_CONTROLLER_INTERVAL)).await;
+        }
+    }
+
+    #[instrument(skip(self, topics_listener))]
+    async fn sync_and_send_topics(
+        &mut self,
+        topics_listener: &mut ChangeListener<TopicSpec, C>,
+    ) -> Result<()> {
+        if !topics_listener.has_change() {
+            debug!("no changes, skipping");
+            return Ok(());
+        }
+
+        let topics = self.ctx.topics().store().clone_values().await;
+        let mirror_topics = topics
+            .into_iter()
+            .map(|topic| {
+                MirroringSpecWrapper::new(topic.key.clone(), topic.spec, topic.status.replica_map)
+            })
+            .filter(|topic| match topic.spec.replicas() {
+                ReplicaSpec::Mirror(MirrorConfig::Home(t)) => t
+                    .partitions()
+                    .iter()
+                    .any(|p| p.remote_cluster == self.req.name),
+                _ => false,
+            })
+            .collect::<Vec<_>>();
```

Comment:
need unit test and need unique identifier so that remote cluster can't get information about other remote clusters

Reply:
we decided to implement TLS, so I think that we don't the unique id.

unit test for the whole sync_and_send_topics?

Reply:
Let's put that as subsequent PR